ydb/library/yql/dq/actors/spilling/compute_storage_actor.cpp

```diff
@@ -0,0 +1,305 @@
+#include "compute_storage_actor.h"
+
+#include "spilling.h"
+#include "spilling_file.h"
+
+#include <ydb/library/yql/utils/yql_panic.h>
+#include <ydb/library/services/services.pb.h>
+
+#include <ydb/library/actors/core/actor_bootstrapped.h>
+#include <ydb/library/actors/core/hfunc.h>
+#include <ydb/library/actors/core/log.h>
+
+#include <util/generic/size_literals.h>
+
+namespace NYql::NDq {
+
+using namespace NActors;
+
+namespace {
+
+#define LOG_D(s) \
+    LOG_DEBUG_S(*TlsActivationContext, NKikimrServices::KQP_COMPUTE, "TxId: " << TxId_ << ". " << s)
+#define LOG_I(s) \
+    LOG_INFO_S(*TlsActivationContext,  NKikimrServices::KQP_COMPUTE, "TxId: " << TxId << ". " << s)
+#define LOG_E(s) \
+    LOG_ERROR_S(*TlsActivationContext, NKikimrServices::KQP_COMPUTE, "TxId: " << TxId_ << ". " << s)
+#define LOG_C(s) \
+    LOG_CRIT_S(*TlsActivationContext,  NKikimrServices::KQP_COMPUTE, "TxId: " << TxId << ". " << s)
+#define LOG_W(s) \
+    LOG_WARN_S(*TlsActivationContext,  NKikimrServices::KQP_COMPUTE, "TxId: " << TxId << ". " << s)
+#define LOG_T(s) \
+    LOG_TRACE_S(*TlsActivationContext,  NKikimrServices::KQP_COMPUTE, "TxId: " << TxId_ << ". " << s)
+
+class TDqComputeStorageActor : public NActors::TActorBootstrapped<TDqComputeStorageActor>,
+                               public IDqComputeStorageActor
+{
+    using TBase = TActorBootstrapped<TDqComputeStorageActor>;
+    // size + promise with key
+    using TWritingBlobInfo = std::pair<ui64, NThreading::TPromise<IDqComputeStorageActor::TKey>>;
+    // remove after read + promise with blob
+    using TLoadingBlobInfo = std::pair<bool, NThreading::TPromise<TRope>>;
+    // void promise that completes when block is removed
+    using TDeletingBlobInfo = NThreading::TPromise<void>;
+public:
+    TDqComputeStorageActor(TTxId txId, const TString& spillerName, std::function<void()>&& wakeupCallback)
+        : TxId_(txId),
+        SpillerName_(spillerName),
+        WakeupCallback_(wakeupCallback)
+    {
+    }
+
+    void Bootstrap() {
+        Become(&TDqComputeStorageActor::WorkState);
+    }
+
+    static constexpr char ActorName[] = "DQ_COMPUTE_STORAGE";
+
+    IActor* GetActor() override {
+        return this;
+    }
+
+    NThreading::TFuture<IDqComputeStorageActor::TKey> Put(TRope&& blob) override {
+        InitializeIfNot();
+        // Use lock to prevent race when state is changed on event processing and on Put call
+        std::lock_guard lock(Mutex_);
+
+        FailOnError();
+
+        ui64 size = blob.size();
+
+        Send(SpillingActorId_, new TEvDqSpilling::TEvWrite(NextBlobId, std::move(blob)));
```

Comment:
Send must be used only inside StateFunc

Reply:
This will be fixed after the YQL-17709. Currently actor may be called outside the actor system (as discussed and as made in ChannelStorageActor(https://github.com/ydb-platform/ydb/blob/70035f45dfabae08d40f703c54cbd55aa060e852/ydb/library/yql/dq/actors/spilling/channel_storage_actor.cpp#L198)

Both actors will be refactored after the YQL-17709.